docs/dotnet/dotnet-network-traces.md

```diff
@@ -0,0 +1,427 @@
+
+DNS spans track logical operations rather than physical DNS calls and the actual behavior depends on the resolver implementation which could be changed in the future versions of .NET.
+.NET 9 uses OS DNS resolver which may do zero or more physical lookups for one API call.
+
+When the *DNS lookup* span is reported along with *HTTP connection setup* and *socket connect* span, the *DNS lookup* span span becomes a child of *HTTP connection setup*
+and a sibling of *socket connect*.
+
+DNS spans are reported for both lookups and reverse lookups.
+
+Lookup (IP addresses from host name) span name SHOULD be `DNS lookup {dns.question.name}`.
+Reverse lookup (host names from IP address) span name SHOULD be `DNS reverse lookup {dns.question.name}`.
+Span kind SHOULD be `INTERNAL`.
+
+Corresponding `Activity.OperationName` is `Experimental.System.Net.NameResolution.DnsLookup`, `ActivitySource` name - `Experimental.System.Net.NameResolution`.
+Added in .NET 9
+
+<!-- semconv span.dotnet.dns.lookup.client -->
+<!-- NOTE: THIS TEXT IS AUTOGENERATED. DO NOT EDIT BY HAND. -->
+<!-- see templates/registry/markdown/snippet.md.j2 -->
+<!-- prettier-ignore-start -->
+<!-- markdownlint-capture -->
+<!-- markdownlint-disable -->
+
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| [`error.type`](/docs/attributes-registry/error.md) | string | The error code or exception name returned by [System.Net.Dns](https://learn.microsoft.com/dotnet/api/system.net.dns). [1] | `host_not_found`; `try_again` | `Conditionally Required` if and only if an error has occurred. | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`dns.answers`](/docs/attributes-registry/dns.md) | string[] | List of resolved IP addresses (for DNS lookup) or a single element containing domain name (for reverse lookup). | `["10.0.0.1", "2001:0db8:85a3:0000:0000:8a2e:0370:7334"]` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`dns.question.name`](/docs/attributes-registry/dns.md) | string | The domain name or an IP address being queried. [2] | `www.example.com`; `opentelemetry.io` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1] `error.type`:** The following errors are reported:
+
+- `host_not_found`
+- `try_again`
+- `no_recovery`
+- `address_family_not_supported`
+- the full exception type name
+
+See [SocketError](https://learn.microsoft.com/dotnet/api/system.net.sockets.socketerror) for more details.
+
+**[2] `dns.question.name`:** If the name field contains non-printable characters (below 32 or above 126), those characters should be represented as escaped base 10 integers (\DDD). Back slashes and quotes should be escaped. Tabs, carriage returns, and line feeds should be converted to \t, \r, and \n respectively.
```

Comment:
I'm not aware of any logic in the implementation that would do the `\DDD` conversion.

Reply:
I think it's a bug in the common attribute definition, removed the note from .net span and created https://github.com/open-telemetry/semantic-conventions/issues/2143